src/Abp.Web/Auditing/WebClientInfoProvider.cs

```diff
@@ -47,11 +47,13 @@ protected virtual string GetClientIpAddress()
 
             var clientIp = httpContext.Request.ServerVariables["HTTP_X_FORWARDED_FOR"] ??
                            httpContext.Request.ServerVariables["REMOTE_ADDR"];
-            
-            clientIp = clientIp.Remove(clientIp.IndexOf(':'));
+
+            // Remove port if present
+            clientIp = clientIp.Contains(":") ? clientIp.Remove(clientIp.IndexOf(':')) : clientIp;
```

Comment:
How about just use `string.Replace(oldChar, newChar)` where oldChar is `’:’` and newChar is `’’`

Reply:
The purpose is to delete the port, just replace ":" is invalid

Reply:
Right, I misunderstood it.

Just personal preference, maybe just need to store the result of `string.IndexOf(“:”)` and check if greater than zero.

E.g
```csharp
var colonIndex = clientIp.IndexOf(“:”);
If (colonIndex >= 0)
{
  clientIp = clientIp.Remove(colonIndex);
}
```
```